substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -2291,6 +2321,66 @@ pub mod pallet {
 			);
 			Ok(())
 		}
+
+		/// Chill an inactive validator.
+		///
+		/// - Anyone can call this function.
+		/// - The validator's stash account should be provided.
+		/// - The proof should be a list of consecutive sorted era indexes where the validator has
+		///   not produced any blocks.
+		/// - This will only be successful if the validator has not produced any blocks in
+		///   [`ChillInactiveValidatorThreshold`] consecutive eras.
+		#[pallet::call_index(30)]
+		#[pallet::weight(T::WeightInfo::chill_inactive_validator(proof.len() as _))]
+		pub fn chill_inactive_validator(
+			origin: OriginFor<T>,
+			stash: T::AccountId,
+			proof: BoundedVec<EraIndex, T::HistoryDepth>,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			ensure!(Validators::<T>::contains_key(&stash), Error::<T>::NotValidator);
+
+			let threshold = ChillInactiveValidatorThreshold::<T>::get();
+
+			ensure!(proof.len() as EraIndex >= threshold, Error::<T>::InvalidProof);
+
+			// Check that the proof is sorted and consecutive.
+			for w in proof.windows(2) {
+				ensure!(w[1] == w[0] + 1, Error::<T>::InvalidProof);
+			}
+
+			let mut consecutive_inactives = 0;
+
+			for era in proof {
+				// Check if the ear exists.
+				if !ErasRewardPoints::<T>::contains_key(era) {
+					consecutive_inactives = 0;
+
+					continue;
```

Comment:
```suggestion
				// Check if the era exists.
				if !ErasRewardPoints::<T>::contains_key(era) {
					break;
```

Invalid proof.

Reply:
Validator inactive for `[0, 1, 2, 3]`.
Threshold `3`.
Era `0` gets clipped from the history at the moment.

This will reject `[0, 1, 2, 3]`. Is this expected?

Reply:
I don't get the question? I mean you only implementing the logic right now, so it doesn't exist yet?